api/src/paths/project/{projectId}/survey/{surveyId}/critters/index.ts

```diff
@@ -0,0 +1,296 @@
+              type: 'array',
+              items: {
+                title: 'marking',
+                type: 'object'
+              }
+            },
+            locations: {
+              title: 'locations',
+              type: 'array',
+              items: {
+                title: 'location',
+                type: 'object'
+              }
+            },
+            mortalities: {
+              title: 'locations',
+              type: 'array',
+              items: {
+                title: 'location',
+                type: 'object'
+              }
+            },
+            qualitative_measurements: {
+              title: 'qualitative measurements',
+              type: 'array',
+              items: {
+                title: 'qualitative measurement',
+                type: 'object'
+              }
+            },
+            quantitative_measurements: {
+              title: 'quantitative measurements',
+              type: 'array',
+              items: {
+                title: 'quantitative measurement',
+                type: 'object'
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  responses: {
+    201: {
+      description: 'Responds with counts of objects created in critterbase.',
+      content: {
+        'application/json': {
+          schema: {
+            title: 'Bulk creation response object',
+            type: 'object'
+          }
+        }
+      }
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/401'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+export function getCrittersFromSurvey(): RequestHandler {
+  return async (req, res) => {
+    const user: ICritterbaseUser = {
+      keycloak_guid: req['system_user']?.user_guid,
+      username: req['system_user']?.user_identifier
+    };
+    const surveyId = Number(req.params.surveyId);
+    const connection = getDBConnection(req['keycloak_token']);
+    const cb = new CritterbaseService(user);
+    const surveyService = new SurveyService(connection);
+    try {
+      await connection.open();
+      const surveyCritters = await surveyService.getCrittersInSurvey(surveyId);
+      const critterIds = surveyCritters.map((a: any) => String(a.critterbase_critter_id));
+      if (!critterIds.length) {
+        return res.status(200).json([]); // This is to catch some unintended behavior in this critterbase endpoint. TODO: Patch Critterbase to handle this correctly.
+      }
+      const result = await cb.filterCritters({ critter_ids: { body: critterIds, negate: false } }, 'detailed');
+
+      for (const bhCritter of surveyCritters) {
+        //This loop is obviously inefficient but probably inconsequential unless people are adding 100+ critters one by one to a survey.
```

Comment:
Should maybe speak to what exactly this method is doing and why, which may help us make it more efficient in the future.